src/Units/UnitContext.java

```diff
@@ -158,6 +169,54 @@ public void setWeapon(WeaponModel w)
     }
   }
 
+  public int getCostTotal()
+  {
+    return (int) (costBase*costMultiplier)+costShift;
+  }
```

Comment:
Before we had `getCost` and `getBuyCost` - is this going to apply a temporary price change to e.g. power accumulation calculations?

Reply:
These values should get modified by any relevant modifiers before this function is called.
`getCost()` and `getBuyCost()` still exist, they just moved out of `UnitModel` since it no longer has all the information to calculate the result.

Temporary price changes still apply, and still should change power accumulation.

Reply:
> still should change power accumulation

Do the carts work that way? Less power accumulation when price discounts are in effect? I don't think we did that before, at any rate.

Reply:
Exactly one cart works that way (AW2), but yes. Hachi (S)COP actually drastically lowers how much charge you can get from killing his units in AW2.
We didn't have anything in place to prevent it before, and I'm pretty sure I tested to verify that we *did* take discounts into account.